src/metrics_writer.py

```diff
@@ -50,6 +51,22 @@ def write_callback(raw_data, data=None):
         met_batcher.push_item(metric)
 
 
+def shutdown_callback():
+    """
+    Shutdown callback. Flushes in memory metrics batches. Default, times out at 5 seconds.
+    """
+    now = time.time()
+    stop = now + met_config.conf[ConfigOptions.shutdown_max_wait]
+    met_batcher.flush()
+    flush_interval = 0.1  # 100 ms
+    # Increase frequency for scheduling http requests
+    met_sender.interval = flush_interval
+    while time.time() < stop and (not met_buffer.empty()):
```

Comment:
After this, the shutdown process should destroy all class instances including metrics_sender I believe. I am happy to add force deleting of the class instances if we believe that is necessary.